libs/sdk-core/src/invoice.rs

```diff
@@ -145,15 +145,20 @@ impl RouteHint {
     }
 }
 
-pub fn add_lsp_routing_hints(
+pub fn add_routing_hints(
     invoice: String,
-    include_route_hints: bool,
-    lsp_hint: Option<RouteHint>,
+    merge_with_existing: bool,
```

Comment:
If there are route hints in the invoice already, and we don't have to replace the route hints because there is no channel needed, do we have to merge in more route hints? Can we keep the ones selected by greenlight and be done with it? Only add route hints if there are none in that case?

Reply:
changed int the caller scope to only include lsp hint if exists.